homeassistant/components/influxdb/const.py

```diff
@@ -0,0 +1,498 @@
+    "Please check that the database, username and password are correct and "
+    "that the specified user has the correct permissions set."
+)
+WRITE_ERROR = "Could not write '%s' to influx due to '%s'."
+QUERY_ERROR = (
+    "Could not execute query '%s' due to '%s'. Check the syntax of your query."
+)
+
+
+# COMMON SCHEMA - Schema and validators used for recorder and sensors
+
+
+def create_influx_url(conf: Dict) -> Dict:
+    """Build URL used from config inputs and default when necessary."""
+    if conf[CONF_API_VERSION] == API_VERSION_2:
+        if CONF_SSL not in conf:
+            conf[CONF_SSL] = DEFAULT_SSL_V2
+        if CONF_HOST not in conf:
+            conf[CONF_HOST] = DEFAULT_HOST_V2
+
+        url = conf[CONF_HOST]
+        if conf[CONF_SSL]:
+            url = f"https://{url}"
+        else:
+            url = f"http://{url}"
+
+        if CONF_PORT in conf:
+            url = f"{url}:{conf[CONF_PORT]}"
+
+        if CONF_PATH in conf:
+            url = f"{url}{conf[CONF_PATH]}"
+
+        conf[CONF_URL] = url
+
+    return conf
+
+
+def validate_version_specific_config(conf: Dict) -> Dict:
+    """Ensure correct config fields are provided based on API version used."""
+    if conf[CONF_API_VERSION] == API_VERSION_2:
+        if CONF_TOKEN not in conf:
+            raise vol.Invalid(
+                f"{CONF_TOKEN} and {CONF_BUCKET} are required when {CONF_API_VERSION} is {API_VERSION_2}"
+            )
+
+        if CONF_USERNAME in conf:
+            raise vol.Invalid(
+                f"{CONF_USERNAME} and {CONF_PASSWORD} are only allowed when {CONF_API_VERSION} is {DEFAULT_API_VERSION}"
+            )
+
+    else:
+        if CONF_TOKEN in conf:
+            raise vol.Invalid(
+                f"{CONF_TOKEN} and {CONF_BUCKET} are only allowed when {CONF_API_VERSION} is {API_VERSION_2}"
+            )
+
+    return conf
+
+
+COMPONENT_CONFIG_SCHEMA_CONNECTION = {
+    # Connection config for V1 and V2 APIs.
+    vol.Optional(CONF_API_VERSION, default=DEFAULT_API_VERSION): vol.All(
+        vol.Coerce(str), vol.In([DEFAULT_API_VERSION, API_VERSION_2]),
+    ),
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Optional(CONF_PATH): cv.string,
+    vol.Optional(CONF_PORT): cv.port,
+    vol.Optional(CONF_SSL): cv.boolean,
+    # Connection config for V1 API only.
+    vol.Inclusive(CONF_USERNAME, "authentication"): cv.string,
+    vol.Inclusive(CONF_PASSWORD, "authentication"): cv.string,
+    vol.Optional(CONF_DB_NAME, default=DEFAULT_DATABASE): cv.string,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    # Connection config for V2 API only.
+    vol.Inclusive(CONF_TOKEN, "v2_authentication"): cv.string,
+    vol.Inclusive(CONF_ORG, "v2_authentication"): cv.string,
+    vol.Optional(CONF_BUCKET, default=DEFAULT_BUCKET): cv.string,
+}
+
+
+# RECORDER SCHEMA - Schema and validators only for recorder
+
+
+def generate_event_to_json(conf: Dict) -> Callable[[Dict], str]:
```

Comment:
This is not a constant. The const.py module should only contain constant.

Reply:
Oh, ok. I thought shared static methods made sense but will remove.

Removing all the refactoring makes sense, will take me a bit though